EntityService/TokenService.php

```diff
@@ -122,11 +123,16 @@ public function setToken(Token $newToken){
                     }
                     // If the channel has the same access token and the same scope,
                     // or no scope has been defined, then we're done.
-                    $status = self::STATUS_SAME_SCOPE;
 
-                    return $status;
+                    return self::STATUS_SAME_SCOPE;
+                } elseif ($newToken->getApplication()->getResourceOwner() === LinkedInController::RESOURCE_OWNER) {
```

Comment:
@ghostika This OAuth module must work independent of any channel-related module. Please implement this check in a generic way, e.g. by adding a second parameter to the setToken() method, e.g. setToken(Token $token, $sameToken = false) and then set $sameToken to true when you call it in the respective Linkedin module.